.github/workflows/ci.yaml

```diff
@@ -0,0 +1,68 @@
+name: "CI"
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master 
+  create:
+    tags:
+      - v*
+
+defaults:
+  run:
+    shell: bash -l {0}
```

Comment:
Since this is somewhat non-standard, let's put a comment on it explaining the purpose of it.

Reply:
Done